app/assets/javascripts/react_components/book.js.jsx

```diff
@@ -188,24 +190,76 @@ var BookList = React.createClass({
+          booklist_ids: $('.selectpicker').val(),
+        },
+        success: function(response) {
+          toastr.success(book.title + " was successfully added to your booklists!");
+        }.bind(this),
+        error: function(xhr, status, err) {
+          var errors = xhr.responseJSON.errors;
+          for (var error of errors) {
+            toastr.error(book.title + " is already in " + error);
+          }
+          console.error(this.props.url, status, err.toString());
+        }.bind(this)
+      });
+    }
+  },
   render: function() {
     bookList = this;
     var bookTiles = this.state.books.map(function (book) {
       return this.generateTile(book);
     }.bind(this));
-
+    var booklists = gon.booklists.map(function(booklist) {
+      return (
+        <option value={booklist.id}>{booklist.name}</option>
+      );
+    }.bind(this));
+    var tagbar_width = (gon.current_user == null) ? "col-md-10" : "col-md-8";
     var searchBar = (
       <div className="row" id="library">
         <div id="tag-and-searchbar">
-          <div className="input-group" id="book-searchbar">
-            <input className="input-block-level form-control" id="book-searchbar-input" onKeyUp={this.search} placeholder="Search for books" type="text" />
-            <span className="input-group-btn">
-              <button className="btn btn-default" id="search-button" onClick={this.search} type="button"><span className="glyphicon glyphicon-search"></span></button>
-            </span>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <div className="input-group" id="book-searchbar">
+                <input className="input-block-level form-control" id="book-searchbar-input" onKeyUp={this.search} placeholder="Search for books" type="text" />
+                <span className="input-group-btn">
+                  <button className="btn btn-default" id="search-button" onClick={this.search} type="button"><span className="glyphicon glyphicon-search"></span></button>
+                </span>
+              </div>
+            </div>
           </div>
-          <div className="input-group" id="book-tagbar">
-            <span className="input-group-addon"><span className="glyphicon glyphicon-tag"></span></span>
-            <input className="input-block-level typeahead form-control" id="book-tagbar-input" placeholder="Add tag" type="text" />
+          <div className="row tagbar-booklists">
+            <div className={tagbar_width + " col-md-offset-1"}>
+              <div className="input-group" id="book-tagbar">
+                <span className="input-group-addon"><span className="glyphicon glyphicon-tag"></span></span>
+                <input className="input-block-level typeahead form-control" id="book-tagbar-input" placeholder="Add tag" type="text" />
+              </div>
+            </div>
+            <div className="select-container">
+              {gon.current_user != null ?
```

Comment:
So I realize that this pattern seems to come up a lot.

I think it might be cleaner to make all of these functions, so we could replace this ternary with `{_renderBooklistDropdown}` and then that function would return the right thing.

Don't worry about this for now but I think it will be worth refactoring later.